### ISSUE TITLE
Rails 6? (NoMethodError: private method `load' called)

### ISSUE THREAD

**Issue by 0xdevalias** [exAspArk/batch-loader#53]
Does this work with Rails 6? I'm following the examples as per the README (using `loader.call`) and instead of getting a `BatchLoader` instance back, i just get `nil`.

Then if i follow the examples as per the original blog post (using `loader.load`), i get a `BatchLoader` instance back, but when I try to access anything on it, i get a private method error:

```
NoMethodError: private method `load' called for #<Proc:0x00007f955766b928>
```

I can't work out if it's not working because i'm doing something wrong, because the docs are wrong, or because it doesn't work with Rails 6, etc.

Rough mockup of my code:

`app/models/foo.rb`
```ruby
# ..snip..

has_many :bars

def bars_lazy
  BatchLoader.for(id).batch do |foo_ids, batch_loader|
    Bar.where(foo_id: foo_ids).each { |s| batch_loader.load(b.id, b) }
  end
end

# ..snip..
```

**Comment by exAspArk** [comment]
Hey @0xdevalias,

I haven't tested it with Rails 6. However, the gem doesn't have any Rails dependencies and ideally should work fine as long as the Ruby version works as expected.

When you call `loader.call` do you pass the correct arguments?

**Comment by petrbela** [comment]
@0xdevalias I'm using it with Rails 6. From your code snippet, it looks like you're trying to call `batch_loader.load` instead of `batch_loader.call`.

**Comment by 0xdevalias** [comment]
I tried .call as per the first line in my original post as well, it didn’t throw an error, but it also didn’t seem to work as expected, unsure if that was an actual issue or me not understanding the docs properly and using it wrong. I’m pretty sure I followed the pattern as described, yet couldn’t make it give me any useful results. From memory when I used the .call pattern it returned nil rather than the expected instance wrapper thing.

Haven’t had a chance to get back to this and explore further.
